src/shipping/components/ShippingWeightUnitForm/ShippingWeightUnitForm.tsx

```diff
@@ -28,46 +23,61 @@ const ShippingWeightUnitForm: React.FC<ShippingWeightUnitFormProps> = ({
 }) => {
   const intl = useIntl();
   const initialForm: FormData = {
-    unit: defaultWeightUnit,
+    unit: defaultWeightUnit
+      ? {
+          label: defaultWeightUnit,
+          value: defaultWeightUnit,
+        }
+      : null,
   };
+  const unitOptions: Option[] = React.useMemo(
+    () =>
+      Object.values(WeightUnitsEnum).map(unit => ({
+        label: unit,
+        value: unit,
+      })),
+    [WeightUnitsEnum],
+  );
 
   return (
     <Form
       confirmLeave
       initial={initialForm}
-      onSubmit={formData => onSubmit(formData.unit)}
+      onSubmit={formData => onSubmit(formData.unit?.value as WeightUnitsEnum)}
```

Comment:
suggestion: `formData.unit` could be potentially null, if so would be better to handle that.